core/src/main/scala/sttp/client4/Response.scala

```diff
@@ -31,38 +31,28 @@ case class Response[+T](
     s"Response($body,$code,$statusText,${Headers.toStringSafe(headers)},$history,$request)"
 }
 
-object Response {
+private[sttp] object Response {
 
-  /** Mainly useful in tests using [[sttp.client4.testing.SttpBackendStub]], when creating stub responses.
-    */
-  val ExampleGet: RequestMetadata = new RequestMetadata {
+  val emptyGet: RequestMetadata = new RequestMetadata {
```

Comment:
I guess we should remove all of these methods from `Response` once `TestResposne` is there?

Reply:
ah sorry, we should only keep the helper methods, but with `requestMetadata: RequestMetadata`. We should move `emptyGet`, though, which kind of is the point of the issue that is being fixed

Reply:
there is a problem here with the methods here as we use them in AbstractStubBackend when request.metadata is not computed yet, I think that we will have to leave it 
![image](https://github.com/softwaremill/sttp/assets/59844835/47de176d-c891-46a1-9ec7-7d36af56486e)

emptyGet should not be available outside sttp package as Respone is package scope, I thought it should be ok from user perspective

Reply:
`emptyGet` or whatever the name should still live in `ResponseStub`/`TestResponse`. The point of the issue is that this shouldn't be exposed in `Response`, but explicitly marked as a test-only thing. In a backend stub, we can of course use a `ResponseStub`